nsqd: nsqlookupd connect callback simpler, more robust

ploxiln opened the pull request:
Remove syncTopicChan and do the initial REGISTER commands in
the lookupPeer connectCallback(). This should be mostly
equivalent, just with less indirection.

Close() the lookupPeer in all error cases. It will attempt to
re-connect with the next Ping command.

This stuff was brought up in #993 - consider this an un-tested proposal. Please question/comment :)

mreiferson commented:
I can't think of any reason why this would be a bad idea!

mreiferson commented:
... mostly this is because I can no longer remember why we wrote it with such indirection. My assumption is that it's just really old Go code of ours that had way too much use of goroutines and channels 😀

ploxiln commented:
Rebased and squashed in the little fixups. I'm going to deploy this out to my infrastructure to let it stew for a bit (but honestly my nsq cluster is not very big or stressed).

ploxiln commented:
I've been running this on a few servers for about a week, using this test-integration branch: https://github.com/ploxiln/nsq/commits/test_201804
